unfurl_message.py

```diff
@@ -260,6 +422,17 @@ def parse_path(path):
 # Serverless handlers
 
 
+def get_handler(path_info):
+    if path_info.note:
+        handler_name = f"get_note_{path_info.type.value}_info"
+    else:
+        handler_name = f"get_{path_info.type.value}_info"
+    handler = globals().get(handler_name)
```

Comment:
Please avoid working with `globals` or `locals` if you don't have very special reason. Especially in combination with constructed function name it seems wrong. There is no apparent bug or security risk in that now, but who knows how the code will evolve.

Explicit `if`, `elif`... is fine. If you want to make it nicer and shorten it, you can define mapping with type and handler combination like:

```python
HANDLERS = {
    PathType.ISSUE: get_issue_info,
    PathType.MERGE_REQUEST: get_mr_info,
    PathType.COMMIT: get_commit_info,
    PathType.PROJECT: get_project_info,
}
```

It's explicit and easy to follow. And strictly limits which functions can be called as handlers.

Reply:
fixed